sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/Command.scala

```diff
@@ -46,5 +47,6 @@ trait AnalysisOnlyCommand extends Command {
   val isAnalyzed: Boolean
   def childrenToAnalyze: Seq[LogicalPlan]
   override final def children: Seq[LogicalPlan] = if (isAnalyzed) Nil else childrenToAnalyze
+  override def innerChildren: Seq[QueryPlan[_]] = if (isAnalyzed) childrenToAnalyze else Nil
```

Comment:
@cloud-fan, updated. Do you also want to explicitly override this to be `Nil` for `CacheTable`, `CacheTableAsSelect` and `UncacheTable`?

Reply:
Does it have real impact in EXPLAIN?

Reply:
There is a change, but I think it's for better:
Before:
```
== Parsed Logical Plan ==
'CacheTableAsSelect tempTable, SELECT key FROM testData, false, false
+- 'Project ['key]
   +- 'UnresolvedRelation [testData], [], false

== Analyzed Logical Plan ==
CacheTableAsSelect tempTable, Project [key#13], SELECT key FROM testData, false, true

== Optimized Logical Plan ==
CacheTableAsSelect tempTable, Project [key#13], SELECT key FROM testData, false, true

== Physical Plan ==
CacheTableAsSelect tempTable, Project [key#13], SELECT key FROM testData, false
```

New:
```
== Parsed Logical Plan ==
'CacheTableAsSelect tempTable, SELECT key FROM testData, false, false
+- 'Project ['key]
   +- 'UnresolvedRelation [testData], [], false

== Analyzed Logical Plan ==
CacheTableAsSelect tempTable, SELECT key FROM testData, false, true
   +- Project [key#13]
      +- SubqueryAlias testdata
         +- View (`testData`, [key#13,value#14])
            +- SerializeFromObject [knownnotnull(assertnotnull(input[0, org.apache.spark.sql.test.SQLTestData$TestData, true])).key AS key#13, staticinvoke(class org.apache.spark.unsafe.types.UTF8String, StringType, fromString, knownnotnull(assertnotnull(input[0, org.apache.spark.sql.test.SQLTestData$TestData, true])).value, true, false) AS value#14]
               +- ExternalRDD [obj#12]

== Optimized Logical Plan ==
CacheTableAsSelect tempTable, SELECT key FROM testData, false, true
   +- Project [key#13]
      +- SubqueryAlias testdata
         +- View (`testData`, [key#13,value#14])
            +- SerializeFromObject [knownnotnull(assertnotnull(input[0, org.apache.spark.sql.test.SQLTestData$TestData, true])).key AS key#13, staticinvoke(class org.apache.spark.unsafe.types.UTF8String, StringType, fromString, knownnotnull(assertnotnull(input[0, org.apache.spark.sql.test.SQLTestData$TestData, true])).value, true, false) AS value#14]
               +- ExternalRDD [obj#12]

== Physical Plan ==
CacheTableAsSelect tempTable, Project [key#13], SELECT key FROM testData, false
```
WDYT?

Reply:
And we can improve the EXPLAIN for physical plans as well as a future PR if needed.

Reply:
yea looks better!